bin/varnishd/http1/cache_http1_deliver.c

```diff
@@ -142,6 +142,9 @@ V1D_Deliver(struct req *req, struct boc *boc, int sendbody)
 
 	if (sc == SC_NULL && err && req->sp->fd >= 0)
 		sc = SC_REM_CLOSE;
-	if (sc != SC_NULL)
+	if (sc != SC_NULL) {
+		VSLb(req->vsl, SLT_DeliveryError, "resp: \"%s\" %d \"%s\"",
+		    sc->desc, errno, VAS_errtxt(errno));
```

Comment:
See 10f1e5cf1dd1042a1f72f49259900888a92bb91d and #4047

Reply:
> I think we should use `sc->name` as with `SessClose` and add the long descriptions from `include/tbl/sess_close.h`

I'm all for logging the names, but against including the descriptions in the manual. "See other section" ought to be enough.

> We should then also remove the quotes.

Once we switch to the name, that goes without saying :)

> See [10f1e5c](https://github.com/varnishcache/varnish-cache/commit/10f1e5cf1dd1042a1f72f49259900888a92bb91d)

I think we should instead add a section dedicated to session/stream close reasons in the manual and add some codegen to generate the RST from the `sess_close.h` table. Then all the tags that print a name can refer to that section and we avoid manual sync.

If you don't, I will.

Reply:
That if statement should be based on whether sc is an error instead of not null.

Reply:
I am happy with all your answers. In particular, moving the session close reasons to a separate docs section makes sense once they are no longer "owned" by `SessClose`.

Reply:
I'm having second thoughts on the tag format and I'm wondering whether we should make it free form like `FetchError` so that any VDP could use `DeliveryError` to log errors?

That does not change my plans to address `SessClose` et al to use the name over the description and add a manual section.

Reply:
How about the prefix (`resp` here) can be replaced by a VDP name and then the format is custom?

Reply:
bugwash: It seems we actually agree on the alternative format `<vdpname>: <error message free form>`